garmin_device_xml.cc

```diff
@@ -127,7 +127,7 @@ const gdx_info*
 gdx_read(const char* fname)
 {
   // Test file open-able before gb_open gets a chance to fatal().
-  FILE* fin = fopen(fname, "r");
+  FILE* fin = ufopen(QString::fromUtf8(fname), "r");
```

Comment:
The comment at line 129 was before xmlgenric was converted to Qt.  Now, instead of having gb_open fatal I think xml_read would fatal in gpsbabel::File::open without a check to make sure the file exists.  When xmlgeneric was converted to Qt we picked up an implicit conversion to QString in xml_init using fromUtf8.  So we already had assumed fname was utf8, but in reality it isn't.  One path goes to get_garmin_mountpoints in jeeps/gpsusbwin.cc which calls GetLogicalDriveStringsA, the ANSI version.  So your PR is no worse than the HEAD, but they probably both fail with certain path names.  I am ready to let that go under the justification that you have done no harm and some good.  This bug has existed for a while without the world collapsing and I would like to move this PR along.

But instead of ufopen I would suggest using QFile::exists(fname).  I think the fromUtf8 conversion will be done implicitly if you don't spell it out which would work out better if gdx_read every takes a QString.  To me the explicit conversion makes me think you know your char string is utf8 and you don't want people to wonder if the implicit conversion is correct, or your getting the code ready for QT_NO_CAST_FROM_ASCII (which would probably be a very long row for us to hoe).  Your thoughts?

Please change the gb_open comment to xml_read.  Please also comment about the potential failure mode described above with a "TODO:"

Reply:
Thanks (I blindly converted the fopen calls without paying particular attention to what they were used for), will fix.